accel-pppd/ipv6/dhcpv6.c

```diff
@@ -752,6 +752,32 @@ static void dhcpv6_recv_rebind(struct dhcpv6_packet *req)
 	dhcpv6_send_reply2(req, pd, D6_REPLY);
 }
 
+static void dhcpv6_recv_confirm(struct dhcpv6_packet *req)
+{
+	struct dhcpv6_pd *pd = req->pd;
+
+	if (!req->clientid) {
+		log_ppp_error("dhcpv6: no Client-ID option\n");
+		return;
+	}
+
+	if (req->serverid) {
+		log_ppp_error("dhcpv6: unexcpected Server-ID option\n");
+		return;
+	}
+
+	if (!pd->clientid)
+		return;
+	else if (pd->clientid->hdr.len != req->clientid->hdr.len || memcmp(pd->clientid, req->clientid, sizeof(struct dhcpv6_opt_hdr) + ntohs(req->clientid->hdr.len))) {
```

Comment:
Are req->clientid->hdr.len and req->clientid->hdr.len validated before this call? If no then there is a risk to access non-allocated memory or garbage